### DISCUSSION TITLE
Help required to add support for Solarbank 2

### DISCUSSION THREAD

**Discussion by thomluther** [thomluther/anker-solix-api#113] [Announcements]
Hi all
As already responded on various support requests for Solarbank 2, I have no device and cannot do any kind of testing to explore Api cabailities with the new Solarbank 2 or smarmeter devices.
Your assistance is needed if you want this support added to the library. Have a look to the [open issues](https://github.com/thomluther/anker-solix-api/issues) and test and explore various endpoints with the new devices. Especially the endpoints to modify any new settings with the solarbank 2. Topics that are unknown:

- **Multiple system_exports are needed to be used for testing and development.** (available, only needed for new/unsupported features)
  - Since I have no real time response data from solarbank 2 or smartmeter devices, only static response data in files can be used for development.
- Smart plug data
  - which data is available in the cloud? Need switch state, timer settings, schedule settings (see #107 and #124)
  - which fields must be provided to apply a change, which are assumed as default
  - which endpoint can be used to apply modifications
- What are the new endpoints `device_attrs `used for, see #107 
- Translation tables for status codes are required
  - what codes and meaning will be found for `grid_status` of smart meter devices reported with grid_info. Only code 0 seen so far, which seems to be 'OK'

**Comment by quazardous** [comment]
Hi I'm interested in your work and I have a e1600 plus (and two old 1600)

Could you create some template script that I could use to scan/test the e1600 plus features ?

I'm a developer but no python experience.

**Reply by thomluther** [reply]
Hi @quazardous 
Actually I don't understand your request.
If you read through the [README](https://github.com/thomluther/anker-solix-api/#anker-solix-api-library), there is the [export_system ](https://github.com/thomluther/anker-solix-api/?tab=readme-ov-file#export_systempy) tool that will extract most of the known Api information from your account.
Most of the features for SB1 and SB2 are understood and implemented meanwhile (will mark it above). Currently I'm working on Smart Plug support see #124 which is quiet new and requires some rework for handling the schedules, since that requires managing 2 different schedules.
What the community still has to explore on its own are unknown information, parameters, endpoints for the Api, such as attributes for Smart Plugs (switch setting, timer, plug schedule etc). I assume endpoint get_device_attrs is used for such features see #107 
All known endpoints (also unused ones) are in [apitypes.py](https://github.com/thomluther/anker-solix-api/blob/main/api/apitypes.py)

**Reply by quazardous** [reply]
Ok thx, I've come across the discussion from Google. Didn't know sb2 were already supported . My bad
Thx
